test/test_nn.py

```diff
@@ -15782,26 +15782,6 @@ def conv2d_depthwise(x, weight):
             with torch.backends.cudnn.flags(enabled=cudnn_enabled):
                 torch.autograd.gradcheck(conv2d_depthwise, (x, weight))
 
-    def _test_batchnorm_grad(self, device, dtype=torch.double):
-        bs, n_feat, size_feat = 4, 5, 6
-        input = torch.arange(bs * n_feat * size_feat, device=device,
-                             requires_grad=True, dtype=dtype).view(bs, n_feat, size_feat)
-        weight = torch.arange(1, n_feat + 1, device=device, requires_grad=True, dtype=dtype)
-        bias = torch.arange(n_feat, device=device, requires_grad=True, dtype=dtype)
-        running_mean = 1 - torch.arange(n_feat, device=device, dtype=dtype)
-        running_var = 2 * torch.arange(n_feat, device=device, dtype=dtype)
-        for training in [False, True]:
-            _assertGradAndGradgradChecks(self, F.batch_norm, (input, running_mean, running_var, weight, bias,
-                                                              training, 0.1, 0.0001))
-
-    def test_batchnorm_grad(self, device):
-        self._test_batchnorm_grad(device)
-
-        if self.device_type == 'cuda' and self.has_cudnn():
-            with torch.backends.cudnn.flags(enabled=False):
-                self._test_batchnorm_grad(device)
```

Comment:
We still want to test that batch norm on CUDA without CUDNN still works. Is there a way to do that using OpInfo?

Reply:
Having another OpInfo with a custom wrapper for the op can be one way, but not sure if it's preferred. Temporarily, I've brought the tests back. Do you think having a separate OpInfo for this variant makes sense?

Apologies for not focusing on this detail.

Reply:
> Apologies for not focusing on this detail.

Reply:
> Having another OpInfo with a custom wrapper for the op can be one way, but not sure if it's preferred. Temporarily, I've brought the tests back. Do you think having a separate OpInfo for this variant makes sense?

@mruberry what say you? I'm down to have a separate OpInfo for this variant (the "definitely not cudnn" variant). The weird thing is that sometimes the cudnn variant and the non-cudnn variant will do the same thing (if we're running on CPU for example, or if a machine just doesn't have cudnn I don't think there is a way to force it)

Reply:
The ideal solution here is probably to add an OpInfo that does something like always applying the following two decorators to its tests:

- @onlyCUDA (so CPU tests are skipped)
- a decorator that disables cuDNN for the duration of the test

A wrapper that disables cuDNN during the call the op is probably OK and could replace the proposed decorator, too.